metrics/repl.lag/pfs.go

```diff
@@ -0,0 +1,269 @@
+// Copyright 2024 Block, Inc.
```

Comment:
Good use of separate file. :-) Iirc, we (or other team members) had this conversation. Usually less files is good, but when some chunk of code really stands alone (in this case, all this PFS logic really does), then a separate pkg files makes editing a little easier. 👍